### ISSUE TITLE
Authorization check

### ISSUE THREAD

**Issue by tibor** [InstaPy/InstaPy#181]
Hi,

with my VPS I ran into the problem that Instagram wanted to check if the request was authorized (e-mail/sms verification).

I wrote this tiny addon which selects e-mail as option at first and then sends the code to Instagram.

```
# placed after regular login script in login_util
# first page where you have to select sms or e-mail as method
email_button = browser.find_elements_by_xpath("//div/div/form/button")[1]

  if email_button is not None:
    action = ActionChains(browser).move_to_element(email_button).click().perform()
    
  sleep(2)

# second page where you would put your verification code     
  input_id = browser.find_element_by_xpath("//div/div/form/input")
  input_send = browser.find_element_by_xpath("//div/div/form/input[@type='submit']")
  if input_id is not None:
    action = ActionChains(browser).move_to_element(input_id).click().send_keys('123456').perform() \
    .move_to_element(input_send).click().perform()

# Call the regular login script again
```

Of cause, we don’t know the code at the first time, so we will have to run the script a second time with the correct code (the code instagram sends stays the same until you have verified your account).

Afterwards you can use this machine not just for this specific account but also for other accounts. For me it worked perfectly.

I haven’t turned it into an PR because it seems to be more an edge case.

Kind regards

Tibor

**Comment by eduardocalazansjr** [comment]
1 - change the find by id, use XPATH.
2 - If the user runs this via cronjob, It will get stuck and the user will never know that.
3 - Double action declaration for a click, why not chain the events?

**Comment by tibor** [comment]
1. will update this later
2. This feature isn’t planned to run via cron but should be thought of as an one time enhancement. When you verify your account once on a machine, this machine will be trusted by Instagram as it seems.
I’m also not sure how to use this the best:
    - Just as a Snippet in the readme.md / FAQ
    - Commented out, but in the regular login_util
    - Used in a debug mode with a parameter like done with ```nogui```
3. You’re right, fixed this. Those were my first lines of Python ever

**Comment by littleswany** [comment]
@tibor I had this issue when I had left my VPN on.

Your idea looks like it would work well.

**Comment by pnoeric** [comment]
I would if this was somehow incorporated into the workflow. Right now I am trying to set up InstaPy on Digital Ocean. Everything works great but of course, it's the first time logging into Instagram account from the DO IP address, so it prompts for the validation code. (I can't see any of this since it is headless, but I receive the SMS with the number.)

So now I have a verification number. How can I enter it? So far the only solution I have found is @tibor's little snippet above, which I'm about to try.

Otherwise, I don't think InstaPy supports this case? So I would be completely stuck? :-(

**Comment by pnoeric** [comment]
BTW just to add - Instagram is a cruel mistress. I tried logging onto the site on my local computer (not using InstaPy, just manually doing it to see the flow). I did this in a browser Private Mode so I would not have a cookie.

I never get the email/SMS option. After entering the username/password, it IMMEDIATELY sends an SMS message and then I land on this screen:

![image](https://user-images.githubusercontent.com/447652/53575793-95f9d700-3b72-11e9-8092-e9c112bf7f07.png)

And what's especially evil is the classes and IDs for that input field and confirm button:

![image](https://user-images.githubusercontent.com/447652/53575817-aa3dd400-3b72-11e9-897f-aab82dfc0377.png)

Not sure how I can get around all this :-)
